packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx

```diff
@@ -0,0 +1,173 @@
+import {Text, vec} from "mafs";
+import * as React from "react";
+
+import {clockwise} from "../../../../util/geometry";
+import {getRayIntersectionCoords as getRangeIntersectionVertex} from "../utils";
+
+import type {CollinearTuple} from "../../../../perseus-types";
+import type {Interval} from "mafs";
+
+interface Props {
+    centerPoint: vec.Vector2;
+    endPoints: [vec.Vector2, vec.Vector2];
+    polygonPoints: readonly vec.Vector2[];
+    active: boolean;
+    range: [Interval, Interval];
+    color?: string;
+}
+
+export const Angle = ({
+    centerPoint,
+    endPoints,
+    range,
+    polygonPoints,
+    color = "var(--movable-line-stroke-color)",
+}: Props) => {
+    const [centerX, centerY] = centerPoint;
+    const areClockwise = clockwise([centerPoint, ...endPoints]);
+    const [[startX, startY], [endX, endY]] = areClockwise
+        ? endPoints
+        : endPoints.reverse();
+
+    const radius = 0.3;
+
+    const a = vec.dist(centerPoint, endPoints[0]);
+    const b = vec.dist(centerPoint, endPoints[1]);
+    const c = vec.dist(endPoints[0], endPoints[1]);
+
+    // Law of cosines
+    const angle = Math.acos((a ** 2 + b ** 2 - c ** 2) / (2 * a * b));
+
+    const startAngle = Math.atan2(startY - centerY, startX - centerX);
+    const endAngle = Math.atan2(endY - centerY, endX - centerX);
+    // const normalizedAngle = angle > 2 * Math.PI ? angle - 2 * Math.PI : angle;
+    const x1 = centerX + radius * Math.cos(startAngle);
+    const y1 = centerY + radius * Math.sin(startAngle);
+    const x2 = centerX + radius * Math.cos(endAngle);
+    const y2 = centerY + radius * Math.sin(endAngle);
+
+    // Fourth point that would create a rhomboid
+    // Used to calculate the inside of the angle and to create square
+    const [x3, y3] = vec.add(
+        centerPoint,
+        vec.add(vec.sub([x1, y1], centerPoint), vec.sub([x2, y2], centerPoint)),
+    );
+
+    // Midpoint betwen ends of arc
+    const isInside = shouldDrawArcInside(
+        [x3, y3],
+        centerPoint,
+        range,
+        polygonPoints,
+    );
+
+    const largeArcFlag = isInside ? 1 : 0;
+    const sweepFlag = isInside ? 1 : 0;
+
+    const arc = `M ${x1} ${y1} A ${radius} ${radius} 0 ${largeArcFlag} ${sweepFlag} ${x2} ${y2}`;
+    const square = `M ${x1} ${y1} L ${x3} ${y3} M ${x3} ${y3} L ${x2} ${y2}`;
+
+    const angleLabelNumber = parseFloat((angle * (180 / Math.PI)).toFixed(1));
+    const angleLabel = Number.isInteger(angleLabelNumber)
+        ? angleLabelNumber
+        : "≈ " + angleLabelNumber;
+
+    return (
+        <>
+            <defs>
+                <filter
+                    id="background"
+                    x="-5%"
+                    width="110%"
+                    y="0%"
+                    height="100%"
```

Comment:
What do these dimensions on the filter do? I tried deleting them and nothing seemed to change...

Reply:
It adds 5% padding on the left and right of the filter.